### PR TITLE
Elamin week-8-practise

### PR DESCRIPTION
**Pull request by aldeez988**

### CODE REVIEW

File: week-8-practise/1-code-refactoring.js [hunk 1]
```diff
@@ -1,35 +1,27 @@
...
+}
 
 // Function that returns text with every other letter capitlised
 export function everyOtherLetterCapitilised(text) {
-  for (i==0; i++; i<text.length) {
-      if (i/2==0) {
-       text[i] = text[i].toUpperCase() 
-    }    
-  } 
+  return text.toUpperCase();
```
Comment (sansaid) [inline week-8-practise/1-code-refactoring.js]: The intention of the function is to make every other letter captilised (i.e. if input is 'practise', output is 'PrAcTiSe'), so the 'for' loop should stay.

However, there is a mistake in the for loop. What about the name of the function?

File: week-8-practise/3-classes.js [hunk 2]
```diff
@@ -23,20 +23,98 @@
...
+    this.location = location;
   }
-
   monthlyEnergyBill(windows) {
-    let base = 40 // base energy bill without windows
-    let multiplier = 1.2 // energy bill expected to go up 20% for each additional window
-    
-    return base + windows*multiplier
+    let base = 40; // base energy bill without windows
+    let multiplier = 1.2; // energy bill expected to go up 20% for each additional window
```
Comment (sansaid) [inline week-8-practise/3-classes.js]: Excellent commenting

File: week-8-practise/3-classes.js [hunk 3]
```diff
@@ -23,20 +23,98 @@
...
+    return base + windows * multiplier;
+  }
+  housePrice() {
+    let baseValue = 50000;
+    let roomsAddedPrice = 10000 * this.numberOfRooms;
+    let totalValue = baseValue + roomsAddedPrice;
+    return this.location === "London" ? (totalValue *= 1.8) : totalValue;
```
Comment (sansaid) [inline week-8-practise/3-classes.js]: Good job at being concise!

File: week-8-practise/3-classes.js [hunk 4]
```diff
@@ -23,20 +23,98 @@
...
- *  - Return and compare the price of the two houses
+ *  - Return and compare the price of the two housese value of the house should increase by 80%
  *  - Return and compare the energy bill of the two houses
  *  - Return how many square meters are in 950 square feet
  */
 
 class House {
-  constructor(windows, colour) {
-    this.windows = windows
+  constructor(windows, colour = "white", numberOfRooms, location) {
```
Comment (sansaid) [inline week-8-practise/3-classes.js]: We haven't covered this in class yet, but default parameters should always be last (JS maps values to the parameters by position, left to right)

File: week-8-practise/3-classes.js [hunk 5]
```diff
@@ -23,20 +23,98 @@
...
+    return this.location === "London" ? (totalValue *= 1.8) : totalValue;
+  }
+
+  static feetIntoMetersSquar(feet) {
+    const oneMetInFootSq = 0.092903;
+    return feet * oneMetInFootSq;
+  }
+}
+
+//comparePrice function receives the houses in an array  as a parameter and compare the price for the two houses
```
Comment (sansaid) [inline week-8-practise/3-classes.js]: Good commenting

File: week-8-practise/3-classes.js [hunk 6]
```diff
@@ -23,20 +23,98 @@
...
+function comparePrice(houses) {
+  if (houses[0].housePrice() > houses[1].housePrice()) {
+    console.log(
+      `The ${houses[0].colour.toUpperCase()} house is £${houses[0].housePrice() -
+        houses[1].housePrice()} more expensive than ${houses[1].colour.toUpperCase()} House`
+    );
+  } else if (houses[0].housePrice() < houses[1].housePrice()) {
+    console.log(
+      `The ${houses[1].colour.toUpperCase()} house is  £${houses[1].housePrice() -
```
Comment (sansaid) [inline week-8-practise/3-classes.js]: Excellent job with template literals!

File: week-8-practise/3-classes.js [hunk 7]
```diff
@@ -23,20 +23,98 @@
...
+var brownHouse = new House(3, "brown", 2, "Woking");
+var blueHouse = new House(5, "blue", 2, "London");
+comparePrice([brownHouse, blueHouse]);
+compareEnergy([brownHouse, blueHouse]);
+
+console.log(
+  `There is ${House.feetIntoMetersSquar(950)} meter squar in 950 foot squar`
+);
+
+/*When we declare a method as static it's not called on the instances(example:brownHouse, blueHouse )
```
Comment (sansaid) [inline week-8-practise/3-classes.js]: Great description

File: week-8-practise/3-classes.js [hunk 8]
```diff
@@ -23,20 +23,98 @@
...
+instead they are called on the class it self in other words we don't need to create an instance of 
+that class to access  the method we can access it directly use class name and the dot notation 
+for example: House.feetIntoMetersSquar(950), if we tried to access a static method using instance
+it will give us an error for example: brownHouse.feetIntoMetersSquar(950)  will generate this error 
+"TypeError: brownHouse.feetIntoMetersSquar is not a function"
+*/
```
Comment (sansaid) [inline week-8-practise/3-classes.js]: Great description

File: week-8-practise/3-classes.js (hunk 4 above)
Comment (aldeez988) [reply week-8-practise/3-classes.js]: Thanks @sansaid  for your review I have updated and added the the default parameter.

File: week-8-practise/1-code-refactoring.js (hunk 1 above)
Comment (aldeez988) [reply week-8-practise/1-code-refactoring.js]: Thanks @sansaid  for your review I have updated and refactor the code again, the name of the function is descriptive but long.

**Pull request closed without merging**